src/is-of-type.ts

```diff
@@ -0,0 +1,68 @@
+import { Action, AnyAction } from './create-action'
+import { ActionCreator } from './create-action-creator'
+import { getType } from './get-type'
+
+export function isOfType<
+  TType extends string | AnyAction | ActionCreator<string>,
+  TAction extends AnyAction
+>(
+  type: TType | TType[],
+  action: TAction
+): action is TAction extends Action<
+  TType extends string
+    ? TType
+    : TType extends AnyAction | ActionCreator<string>
+    ? TType['type']
+    : never
+>
+  ? TAction
+  : never
+
+export function isOfType<
+  TType extends string | AnyAction | ActionCreator<string>
+>(
+  type: TType | TType[]
+): <TAction extends AnyAction>(
+  action: TAction
+) => action is TAction extends Action<
+  TType extends string
+    ? TType
+    : TType extends AnyAction | ActionCreator<string>
+    ? TType['type']
+    : never
+>
+  ? TAction
+  : never
+
+export function isOfType<
+  TType extends string | AnyAction | ActionCreator<string>,
+  TAction extends AnyAction
+>(typeOrTypes: TType | TType[], action?: TAction) {
+  const types = Array.isArray(typeOrTypes) ? typeOrTypes : [typeOrTypes]
+
+  const isType = <TType>(type: unknown): type is TType & string =>
+    typeof type === 'string'
+  const isActionOrActionCreator = <TType>(
+    type: any
+  ): type is TType & (AnyAction | ActionCreator<string>) =>
+    (typeof type === 'object' || typeof type === 'function') &&
+    !!(type && type.type)
```

Comment:
`isType` and `isActionOrActionCreator` functions are pure. So it's better to move them to utils. If it's important to change their type you can cast their types inside `isOfType` function.

Reply:
completed!